orbit_viewer/_intervals.py

```diff
@@ -0,0 +1,74 @@
+import PySide2.QtCore as QtCore
+
+import datetime as dt
+
+from typing import List, Tuple
+
+from dataclasses import dataclass, field
+
+# TODO if interval A has been selected (in list 's') and range is changed so that A is out-of-range we keep it in
+# the list of selected intervals to have it still selected if we are coming back to this range.
+# If the interval does not exists anymore (due to filter-changes) the selection is still true
+# -> needs some housekeeping
+
+
+@dataclass
+class _IntervalsData:
+    i: List[Tuple[dt.datetime, dt.datetime]] = field(default_factory=set)
+    s: List[Tuple[dt.datetime, dt.datetime]] = field(default_factory=set)
+
+
+class Intervals(QtCore.QObject):
+    """ Model containing all intervals of all trajectories """
+
+    intervals_changed = QtCore.Signal(str)
+    trajectory_deleted = QtCore.Signal(str)
+    selection_changed = QtCore.Signal(str)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
not sure about this, QObject ctor normally expect a QObject parent or None.

Reply:
I understood it as "all constructor-overloads supported by the parent-class" - without further thinking.